geonode/security/middleware.py

```diff
@@ -169,28 +169,20 @@ def __init__(self, get_response):
     def process_request(self, request):
         whitelist = getattr(settings, 'ADMIN_IP_WHITELIST', [])
         if len(whitelist) > 0:
-            user = None
-            if getattr(request, "user", None):
-                user = request.user
-            else:
-                user = extract_user_from_headers(request)
-            if user and user.is_superuser:
+            potential_admins = []
+            potential_admins.append(extract_user_from_headers(request))
```

Comment:
I'll comment here on why this is needed and which are the cases. Just to document it